### PR TITLE
Converting Mockito to MockK library

### PR DESCRIPTION
**Pull request by oleksiyp**
I think diff is quite self-explanatory.
Going to write an article on how to convert mockito to my lib.

### CODE REVIEW

File: imagefeedview/src/test/java/com/oliveroneill/imagefeedview/adapter/PhotoPagerAdapterTest.kt [hunk 1]
```diff
@@ -18,30 +14,23 @@ class PhotoPagerAdapterTest {
     lateinit var adapter : PhotoPagerAdapter<String>
     lateinit var mockController : ImageFeedController<String>
     lateinit var mockView : PhotoPagerAdapter.ViewHolder
-    lateinit var mockImageView : GestureImageView
     @Before
     fun setup() {
-        mockController = mock()
+        mockController = mockk(relaxed = true)
```
Comment (oleksiyp) [inline imagefeedview/src/test/java/com/oliveroneill/imagefeedview/adapter/PhotoPagerAdapterTest.kt:19]: here you can see that all replaced call chains are mocked automatically i.e. `mock1.call1` is a mockk and `mock1.call1.call2` is a mockk, e.t.c.

File: imagefeedview/src/test/java/com/oliveroneill/imagefeedview/ImageFeedViewTest.kt [hunk 2]
```diff
@@ -20,14 +18,14 @@ class ImageFeedViewTest {
...
     fun setup() {
-        mockGrid = mock()
-        mockPager = mock()
-        feed = ImageFeedView(mock())
-        feed.setGridAdapter(mockGrid, mock())
-        feed.setPagerAdapter(mockPager, mock())
-        feed.initGridAdapter(mock(), 3)
-        feed.initPagerAdapter(mock())
-        feed.setSwipeRefresh(mock())
+        mockGrid = mockk(relaxed = true)
```
Comment (oleksiyp) [inline imagefeedview/src/test/java/com/oliveroneill/imagefeedview/ImageFeedViewTest.kt:21]: relaxed mock required if you are not going to specify mock behavior explicitly, by default mocks are strict

File: imagefeedview/src/test/java/com/oliveroneill/imagefeedview/adapter/PhotoPagerAdapterTest.kt (hunk 1 above)
Comment (oliveroneill) [reply imagefeedview/src/test/java/com/oliveroneill/imagefeedview/adapter/PhotoPagerAdapterTest.kt:19]: That's a really nice feature!

**Comment by oliveroneill** [comment]
Nice work! Thanks so much!

**Pull request merged**